### ISSUE TITLE
Unescaped left brace in regex is deprecated, passed through in regex

### ISSUE THREAD

**Issue by jonasstein** [ftilmann/latexdiff#107]
```
$ latexdiff --version
This is LATEXDIFF 1.2.0  (Algorithm::Diff 1.15 so, Perl v5.22.1)
  (c) 2004-2016 F J Tilmann
```

```
$ latexdiff A.tex B.tex > diff.tex
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/command{ <-- HERE \\DIFaddbeginFL}{((?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}])*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*)}/ at /usr/local/bin/latexdiff line 1171, <DATA> line 31168.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/command{\\DIFaddbeginFL}{ <-- HERE ((?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}]|\{(?:[^{}])*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*\})*)}/ at /usr/local/bin/latexdiff line 1171, <DATA> line 31168.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/\\begin{ <-- HERE ((?:picture|tikzpicture|DIFnomarkup)[\w\d*@]*)}(.*?)\\end{\1}/ at /usr/local/bin/latexdiff line 2354, <DATA> line 35785.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/\\begin{((?:picture|tikzpicture|DIFnomarkup)[\w\d*@]*)}(.*?)\\end{ <-- HERE \1}/ at /usr/local/bin/latexdiff line 2354, <DATA> line 35785.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/\\begin{ <-- HERE ((?:aligned|array|[pbvBV]?matrix|smallmatrix|cases|split))}(.*?)\\end{\1}/ at /usr/local/bin/latexdiff line 2359, <DATA> line 35785.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/\\begin{((?:aligned|array|[pbvBV]?matrix|smallmatrix|cases|split))}(.*?)\\end{ <-- HERE \1}/ at /usr/local/bin/latexdiff line 2359, <DATA> line 35785.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/\\begin{ <-- HERE ((?:equation[*]?|displaymath|DOLLARDOLLAR)[*]?|(?:eqnarray|align|alignat|gather|multline|flalign)[*]?|SQUAREBRACKET)}(.*?)\\frac(([\s]*%[^\n]*?)*[\r\n|\r|\n])+\{(.*?)\\end{\1}/ at /usr/local/bin/latexdiff line 2258, <DATA> line 35785.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/\\begin{((?:equation[*]?|displaymath|DOLLARDOLLAR)[*]?|(?:eqnarray|align|alignat|gather|multline|flalign)[*]?|SQUAREBRACKET)}(.*?)\\frac(([\s]*%[^\n]*?)*[\r\n|\r|\n])+\{(.*?)\\end{ <-- HERE \1}/ at /usr/local/bin/latexdiff line 2258, <DATA> line 35785.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/\\begin{ <-- HERE ((?:equation[*]?|displaymath|DOLLARDOLLAR)[*]?|(?:eqnarray|align|alignat|gather|multline|flalign)[*]?|SQUAREBRACKET)}(.*?)\\end{\1}/ at /usr/local/bin/latexdiff line 2363, <DATA> line 35785.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/\\begin{((?:equation[*]?|displaymath|DOLLARDOLLAR)[*]?|(?:eqnarray|align|alignat|gather|multline|flalign)[*]?|SQUAREBRACKET)}(.*?)\\end{ <-- HERE \1}/ at /usr/local/bin/latexdiff line 2363, <DATA> line 35785.

```

**Comment by jonasstein** [comment]
thank you. I will test it. With released this weekend you mean released via texlive?

**Comment by ftilmann** [comment]
No, released on github and on www.ctan.org.   When this will be incorporated into texlive is beyond my control and you will have to ask texlive maintainers about that

**Comment by misho104** [comment]
Hi, one left-bracket is still left:
```
@@ -2864,7 +2855,7 @@
       s/\\begin\{MATHMODE\}(.*?)\\end\{MATHMODE\}/\\begin{$MATHARRREPL}$1\\end{$MATHARRREPL}/sg;

       # now look for AUXCMD math-mode pairs which have only comments (or empty lines between them), and remove the added commands
-      s/\\begin\{((?:$MATHENV)|(?:$MATHARRENV)|SQUAREBRACKET)}$AUXCMD\n((?:\s*%.[^\n]*\n)*)\\end\{\1}$AUXCMD\n/$2/sg;
+      s/\\begin\{((?:$MATHENV)|(?:$MATHARRENV)|SQUAREBRACKET)}$AUXCMD\n((?:\s*%.[^\n]*\n)*)\\end{\1}$AUXCMD\n/$2/sg;
     } else {
       #   math modes OFF,WHOLE,COARSE: Convert \MATHBLOCKmath{..} commands back to environments
       s/\\MATHBLOCK($MATHENV|$MATHARRENV|SQUAREBRACKET)\{($pat_n)\}/\\begin{$1}$2\\end{$1}/sg;
```

**Comment by ftilmann** [comment]
Thanks, fixed in commit 61ee64f
